impeller/compiler/spirv_sksl.cc

```diff
@@ -350,7 +374,9 @@ void CompilerSkSL::emit_resources() {
     statement("");
   }
 
-  declare_undefined_values();
+  if (emit_undefined_values()) {
+    statement("");
+  }
```

Comment:
`declare_undefined_values` was removed by https://github.com/KhronosGroup/SPIRV-Cross/commit/5547b25afe15a1e4c47858563fd952d25d8858be.

For SPIRV's GLSL compiler, undefined values are emitted by `CompilerGLSL::emit_resources`. Previously `emit_resources` called `CompilerGLSL::declare_undefined_values`; now `emit_resources` itself emits undefined values without any helpers.

Since `CompilerGLSL::declare_undefined_values` was removed, our SkSL compiler has two options to emit undefined values:

1. Copy the old logic from `CompilerGLSL::declare_undefined_values` into `SkSL`, or...
2. Refactor `CompilerSkSL` to call `CompilerGLSL::emit_resources`

I went with the first option as that's the easiest.